st2actions/st2actions/scheduler/entrypoint.py

```diff
@@ -61,28 +78,20 @@ def process(self, request):
             LOG.exception('Failed to find liveaction %s in the database.', request.id)
             raise
 
-        # Apply policies defined for the action.
-        liveaction_db = policy_service.apply_pre_run_policies(liveaction_db)
-
-        # Exit if the status of the request is no longer runnable.
-        # The status could have be changed by one of the policies.
-        if liveaction_db.status not in [action_constants.LIVEACTION_STATUS_REQUESTED,
-                                        action_constants.LIVEACTION_STATUS_SCHEDULED]:
-            LOG.info('%s is ignoring %s (id=%s) with "%s" status after policies are applied.',
-                     self.__class__.__name__, type(request), request.id, liveaction_db.status)
-            return
-
-        # Update liveaction status to "scheduled".
-        if liveaction_db.status == action_constants.LIVEACTION_STATUS_REQUESTED:
+        if liveaction_db.delay > 0:
             liveaction_db = action_service.update_status(
-                liveaction_db, action_constants.LIVEACTION_STATUS_SCHEDULED, publish=False)
+                liveaction_db,
+                action_constants.LIVEACTION_STATUS_DELAYED,
+                publish=False
+            )
```

Comment:
Looking at the rest of the entrypoint code, it will add the liveaction to the scheduling collection. It is not scheduled at this point. Only processed to await scheduling by the handler. If there is a delay it will add the delay to the scheduled start time and set the liveaction to `delayed` until the scheduler updates it to `scheduled`

Reply:
Yeah, I'm not finding the line where the liveaction is set to scheduled.

Reply:
It happens in the pre_run_policies see: https://github.com/StackStorm/st2/pull/4397#discussion_r232081923